WorkloadManagementSystem/Agent/SiteDirector.py

```diff
@@ -626,7 +685,16 @@ def submitJobs( self ):
           return result
 
         executable, pilotSubmissionChunk = result['Value']
-        result = ce.submitJob( executable, '', pilotSubmissionChunk )
+       
+        #check the parameter
+        processors = self.pilotProcessors
+        if self.wholeNode == 'true':
+          processors = -1
+        if processors != 1:
```

Comment:
Calling `ce.submitJob( executable, '', pilotSubmissionChunk, processors = processors )` directly without the test should be better?